### ISSUE TITLE
Help/user navigation accessibility issues

### ISSUE THREAD

**Issue by seanforyou23** [syndesisio/syndesis#2201]
## This is a...

<pre><code>
[ ] Feature request
[ ] Regression (a behavior that used to work and stopped working in a new release)
[x] Bug report  
[ ] Documentation issue or request
</code></pre>

## The problem

When navigating syndesis application with a screen reader (VoiceOver for example), it becomes obvious there are some shortcomings in our accessibility support. Namely, the help and user account dropdowns do not expose a human readable text, but instead rely on icons which aren't helpful to screen reader users. Further, their expanded/collapsed state isn't reflected and high-level page landmarks are not always reflected.

## Expected behavior

All main navigational links that use icons instead over text have a text equivalent which is exposed via aria-label attribute.

Dropdowns should be linked to their parent menu item via aria-labeledby attribute.

Parent menu items should be connected to their dopdown containers via aria-controls attribute.

All page landmarks should be exposed by explicitly defining role's for the primary page regions.

## Screenshot

![broken01](https://user-images.githubusercontent.com/5942899/38096424-d5bd411e-3340-11e8-9a75-979bbfd99c71.jpg)
![broken02](https://user-images.githubusercontent.com/5942899/38096425-d5cacfb4-3340-11e8-8599-4e1b61354d12.jpg)

## Tasks involved / Steps to Reproduce

1. Launch the VoiceOver screen reader (cmd + F5)
2. Pull up page links in the rotor menu (VO + U)
3. Notice the help and user navigation items are not readable
4. Also notice the help/user navigation items do not reflect expanded/collapsed state when read aloud
